lib/syscall_shim/gen_provided.awk

```diff
@@ -7,7 +7,11 @@ BEGIN {
 
 /[a-zA-Z0-9]+-[0-9]+/{
 	printf "\n#ifndef SYS_%s\n", $1;
-	printf "#warning Ignoring system call '%s': No system call number available\n", $1
+	printf "#ifdef LEGACY_%s\n", $1;
```

Comment:
Could you add a short explanation of what this does? If the `LEGACY_...` macro is defined then that means there is a legacy system call that is ignored?

So a system call is either:
* required and defined: all OK
* not required and not-defined (i.e. legacy), warning shown
* required and not-defined, error shown

Is this correct?

Reply:
Yes, exactly. I will add the explanation.